src/redisearch_rs/inverted_index/src/index_result.rs

```diff
@@ -251,6 +251,19 @@ impl<'index> RSTermRecord<'index> {
             },
         }
     }
+
+    /// Set the offsets of this term record, replacing any existing offsets.
+    pub fn set_offsets(&mut self, offsets: RSOffsetVector<'index>) {
+        match self {
+            RSTermRecord::Borrowed { offsets: o, .. } => {
+                *o = offsets;
+            }
+            RSTermRecord::Owned { offsets: o, .. } => {
+                o.free_data();
+                *o = offsets.to_owned();
```

Comment:
I guess there is no case where we should change the ownership tag?

Reply:
No, there is no case since we are either setting borrowed offsets to the borrowed type, or making an owned copy when setting it to the owned type.